message_downloader.py

```diff
@@ -98,8 +97,10 @@ def download(self, start_at = None):
                             cid, message['thread_ts'])
                         thread_messages = [
                             x for x in thread_messages if x['thread_ts'] != x['ts']]
+                        message['replies'] = thread_messages
                         self.MessageWriter.write(
                             thread_messages, cid, parent_user_id)
+                self.MessageWriter.write(messages, cid)
```

Comment:
Can you share more about how this fixes the bug?

Reply:
The way we keep track of the replies to a message is to have a field in the message called 'replies' which contains a list of replies to the message.  

Now, I'm not exactly sure what was different a month ago, but right now, the problem we had was that we would:
1. Get the list of messages for a channel (which would not include threaded messages); 
2. Save those messages to the database;
3. For each that were a threadhead, get the messages in the thread they started
4. Save those messages to the database

So we were accurately capturing all the messages, but we were not capturing the RELATIONSHIP between them.  

So instead, what we're doing now is 
1. Get the list of messages for a channel (which would not include threaded messages); 
2. For each that's a threadhead, get thread messages and save those in DB
3. Set the 'replies' field for each threadhead to the thread messages
4. Save the top-level messages (now with a 'replies' field) to the DB